Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServerTest.cs

```diff
@@ -54,16 +54,28 @@ public void Ctor_LogsServerTypeAndVersion()
     }
 
     [DataTestMethod]
-    [DataRow("7.9.0.5545", false)]
-    [DataRow("8.0.0.18670", false)]
-    [DataRow("8.8.0.1121", false)]
-    [DataRow("8.9.0.0", true)]
-    [DataRow("9.0.0.1121", true)]
-    [DataRow("10.15.0.1121", true)]
-    public void IsServerVersionSupported(string sqVersion, bool expected)
+    [DataRow("7.9.0.5545", false, false)]
+    [DataRow("8.0.0.18670", false, false)]
+    [DataRow("8.8.0.1121", false, false)]
+    [DataRow("8.9.0.0", true, true)]
+    [DataRow("9.0.0.1121", true, true)]
+    [DataRow("9.9.0.0", true, false)]
+    [DataRow("10.15.0.1121", true, false)]
+    public void IsServerVersionSupported(string sqVersion, bool expected, bool shouldLogWarning)
     {
-        var sut = CreateServer(version: new Version(sqVersion));
+        var logger = new TestLogger();
+        var sut = CreateServer(version: new Version(sqVersion), logger: logger);
         sut.IsServerVersionSupported().Should().Be(expected);
+        if (!expected)
+        {
+            logger.AssertErrorLogged("SonarQube versions below 8.9 are not supported anymore by the SonarScanner for .NET. Please upgrade your SonarQube version to 8.9 or above or use an older version of the scanner (< 6.0.0), to be able to run the analysis.");
+        }
+        if (shouldLogWarning)
+        {
+            var expectedWarningMessage = "SonarQube versions below 9.9 will be unsupported by the SonarScanner for .NET starting from January 2025. Please upgrade to a newer SonarQube version.";
+            logger.AssertWarningLogged(expectedWarningMessage);
+            logger.AssertUIWarningsLogged(expectedWarningMessage);
+        }
```

Comment:
I prefer to keep the tests as simple as possible. I would prefer to have multiple tests that will assert the messages instead of packing everything in one single test.